### PR TITLE
Add option to skip head on snapshots

### PR DESCRIPTION
**Pull request by fabxc**
@brian-brazil 

Since we (for reasons still unclear to me) back-ported the snapshot stuff to /v1 I actually had to invert the flag. Feels a bit inconsistent but seems like the best way for now.

### CODE REVIEW

**Comment by fabxc** [comment]
@SuperQ Codacy back again from your account it seems. What kind of cancer is this =/

**Comment by SuperQ** [comment]
@fabxc Yea, it's pretty awful.  Are we trying to keep this Codacy thing, should I try and disconnect it?

**Comment by brian-brazil** [comment]
We've been trying to get rid of it for a while.

**Comment by brian-brazil** [comment]
Can you update the docs?

**Comment by fabxc** [comment]
Done.

File: docs/querying/api.md [hunk 1]
```diff
@@ -452,9 +452,10 @@ We also expose a gRPC API whose definition can be found [here](https://github.co
 
 ### Snapshot
 Snapshot creates a snapshot of all current data into `snapshots/<datetime>-<rand>` under the TSDB's data directory and returns the directory as response.
+It will optionally skip snapshotting data that is only present in memory and the write ahead log.
```
Comment (brian-brazil) [inline docs/querying/api.md]: This is hard to understand. Maybe: skip snapshotting data that is only present in the head block, and which has not yet been written out as a block.

**Comment by fabxc** [comment]
done

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**